repo.go

```diff
@@ -309,19 +309,29 @@ func (r *Repo) GenKey(role string) ([]string, error) {
 }
 
 func (r *Repo) GenKeyWithExpires(keyRole string, expires time.Time) (keyids []string, err error) {
-	// Not compatible with delegated targets roles, since delegated targets keys
-	// are associated with a delegation (edge), not a role (node).
+	return r.GenKeyWithTypeAndExpires(keyRole, expires, data.KeyTypeEd25519)
+}
 
-	signer, err := keys.GenerateEd25519Key()
-	if err != nil {
-		return []string{}, err
+func (r *Repo) GenKeyWithTypeAndExpires(role string, expires time.Time, keyType data.KeyType) ([]string, error) {
+	var signer keys.Signer
+	var err error
+	switch keyType {
+	case data.KeyTypeEd25519:
+		signer, err = keys.GenerateEd25519Key()
+	case data.KeyTypeECDSA_SHA2_P256:
+		signer, err = keys.GenerateEcdsaKey()
+	case data.KeyTypeRSASSA_PSS_SHA256:
```

Comment:
This is where I think things get confusing. I understand that key types and schemes are distinct, but if we are only going to support "rsa-pss-sha256", then we should use schemes instead of types here to make things sparkling clear.

Reply:
I agree, changing to scheme.